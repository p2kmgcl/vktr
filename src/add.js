import haveSameNumberOfDimensions from "./assertions/have-same-number-of-dimensions";
import isVector from "./assertions/is-vector";
import map from "./map";

/**
 * Returns a new vector result of (a + b)
 * @param {Vector} a
 * @param {Vector} b
 * @return {Vector}
 */
function add(a, b) {
  isVector(a);
  isVector(b);
  haveSameNumberOfDimensions(a, b);

  return map(a, (component, index) => component + b.components[index]);
}

export default add;
