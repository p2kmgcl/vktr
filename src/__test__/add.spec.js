import add from "../add";
import Vector from "../vector";

describe("add()", () => {
  it("allows adding two vectors", () => {
    const a = new Vector(1);
    const b = new Vector(2);

    expect(add(a, b)).toMatchSnapshot();
  });

  it("requires vector parameter", () => {
    const a = new Vector(1);

    expect(() => add(a, 1)).toThrow(TypeError);
  });

  it("requires same number of dimensions", () => {
    const a = new Vector(1);
    const b = new Vector(1, 2);

    expect(() => add(a, b)).toThrow(RangeError);
  });
});
