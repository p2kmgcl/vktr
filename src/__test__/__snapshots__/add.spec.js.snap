// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add() allows adding two vectors 1`] = `
Array [
  3,
]
`;
